src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1120,7 +1119,9 @@ public override BoundNode VisitConversion(BoundConversion conversion)
                 if (conversion.ConversionKind == ConversionKind.MethodGroup &&
                     conversion.SymbolOpt?.MethodKind == MethodKind.LocalFunction)
                 {
-                    return conversion;
+                    return new PartiallyLoweredLocalFunctionReference(
+                        conversion,
+                        new Dictionary<Symbol, CapturedSymbolReplacement>(proxies));
```

Comment:
@cston The proxies list is not immutable, though, and things are actually added and removed from it. I'm not sure what code you're suggesting to deal with that in the visitors.